harness/internal/acp/session_cost_test.go

```
@@ -0,0 +1,340 @@
+	}()
+
+	s.next() // the session/prompt request
+
+	// Simulate goose hitting its native turn limit mid-generation: some
+	// real tool_call notifications arrive, then the connection drops with
+	// no final response — never a graceful stopReason.
+	for i := 0; i < 3; i++ {
+		s.push(`{"jsonrpc":"2.0","method":"session/update","params":{"sessionId":"s1","update":{"sessionUpdate":"tool_call"}}}`)
+	}
+	s.mu.Lock()
+	s.conn.Close()
```

golangci-lint (2.12.2)

[error] Error return value of `s.conn.Close` is not checked

(errcheck)